tests/rptest/services/cluster.py

```diff
@@ -53,7 +53,8 @@ def wrapped(self, *args, **kwargs):
                     # TODO: extend this to cover shutdown logging too, and
                     # clean up redpanda to not log so many errors on shutdown.
                     self.redpanda.raise_on_bad_logs(allow_list=log_allow_list)
-                    return r
+                self.redpanda.trim_logs()
```

Comment:
I did move this line out of the conditional. I however don't see a difference because in the case `raise_on_bad_logs` throws, the exception would abruptly stop the method and `r` would never be returned.